api/features/tasks.py

```diff
@@ -82,3 +192,82 @@ def _get_feature_state_webhook_data(feature_state, previous=False):
         identity_identifier=getattr(feature_state.identity, "identifier", None),
         feature_segment=feature_state.feature_segment,
     )
+
+
+def _save_feature_state_value_with_type(
+    value: Optional[Union[int, bool, str]], feature_state_value: FeatureStateValue
+) -> None:
+    if isinstance(value, int):
+        feature_state_value.type = INTEGER
+        feature_state_value.integer_value = value
+    elif isinstance(value, bool):
+        feature_state_value.type = BOOLEAN
+        feature_state_value.boolean_value = value
+    else:
+        feature_state_value.type = STRING
+        feature_state_value.string_value = value
+
+    feature_state_value.save()
+
+
+def _create_multivariate_feature_option(
```

Comment:
I think this could be simplified by doing something like: 

```python
mvfo = MultivariateFeatureOption(feature=feature, default_percentage_allocation=default_percentage_allocation)
if isinstance(value, int):
    mvfo.integer_value = value
    mvfo.type = INTEGER
...
```

Reply:
Cool. I did something kinda similar where we load the type from the now saved type in the data, with an assert at the end to verify that the type is one of the three known types.